src/components/PopupWithForm.js

```diff
@@ -15,16 +19,21 @@ export default class PopupWithForm extends Popup {
         });
         return this._formValues;
     }
-
+    setLoadingState(isLoading) {
+        if (this._submitButton) {
+          this._submitButton.textContent = isLoading
+          ? "Saving..."
+          : this._defaultButtonText;

```

Comment:
**COULD BE IMPROVED**

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```